### PR TITLE
Update PostAdaptiveCard to latest version

### PR DESCRIPTION
**Pull request by MiYanni**
Updated to the latest version. It was recommended by the .NET OSS Admins to use the commit SHA as the version number.

Edit: It worked because this PR was shown.
image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7546)

### CODE REVIEW

File: .github/workflows/notify-teams-channel.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Post PRs and issues to Teams
+name: notify-teams-channel
```
Comment (MiYanni) [inline .github/workflows/notify-teams-channel.yml:1]: Updated the name that is displayed in the GitHub Checks area to match the naming convention we have for the other checks.

File: .github/workflows/notify-teams-channel.yml [hunk 2]
```diff
@@ -7,12 +7,13 @@ on:
...
 jobs:
-  notify:
+  notify-teams-channel:
+    name: post
     runs-on: ubuntu-latest
     
     steps:
       - name: Notify
-        uses: davidwengier/PostAdaptiveCard@v1.0.4
+        uses: davidwengier/postadaptivecard@ed077f2bd4b80fdaaef21dae2e13a34e556d466d
```
Comment (swesonga) [inline .github/workflows/notify-teams-channel.yml:16]: Where does this SHA come from?

Comment (MiYanni) [reply .github/workflows/notify-teams-channel.yml:16]: It is the commit SHA that relates to that version of the action. I mentioned the lack of user-friendliness of accessing this in my email thread with the .NET OSS Admins.

**Review by tmeschter — APPROVED** [review]

**Review by jjmew — APPROVED** [review]
:shipit:

File: .github/workflows/notify-teams-channel.yml [hunk 3]
```diff
@@ -1,18 +1,19 @@
-name: Post PRs and issues to Teams
+name: notify-teams-channel
 
 on:
   pull_request:
-    types: [opened, reopened]
+    types: [opened, reopened, ready_for_review]
```
[the remark this answers is not in the document]
Comment (MiYanni) [reply .github/workflows/notify-teams-channel.yml:5]: Yeah. I noticed that draft PRs might remain for a couple weeks, depending on situations like a set of CPS changes that the PR is waiting on. So, I believe putting a new card in the channel when the PR becomes 'real' makes sense. Unfortunately, there isn't a way to have `opened` only represent non-draft PRs. That would be ideal so we'd only get 1 post..

**Pull request merged**